### PR TITLE
feat: add --size flag to zellij run for specifying split size

### LINKED ISSUE #2804: feat: specify size for `zellij run`
Additionally to the `-d, --direction <DIRECTION>` for `zellij run` it would be nice to have a `--size` flag that determines the size for the openend split.

### PR DESCRIPTION
**Pull request by veeceey**
## Summary

Adds a `--size` flag to `zellij run` and `zellij action new-pane` for specifying the size of the new pane when splitting.

```bash
zellij run -d down --size 20%     # new pane takes 20% of available space
zellij run -d right --size 30%
zellij action new-pane -d down --size 50%
```

`--size` requires a direction (`-d`) and conflicts with `--floating`, `--in-place`, and `--stacked`.

## Changes

- **CLI**: Added `--size` flag to both `Sessions::Run` and `CliAction::NewPane`
- **Data model**: Added `size: Option<PercentOrFixed>` to `NewPanePlacement::Tiled`
- **Tab/Pane splitting**: Added `split_with_size()` in `tiled_pane_grid.rs` for non-equal splits, with `horizontal_split_with_size` and `vertical_split_with_size` on `Tab`
- **Protobuf/Plugin API**: Updated serialization for backward compatibility

## Test plan

- [x] 9 unit tests for `split_with_size()` covering equal splits, percentage splits, coordinate preservation, clamping, and edge cases
- [x] All existing tests pass (`cargo test -p zellij-server`)
- [x] Manual testing: `zellij run -d down --size 20%` creates a pane taking 20% of available space

Closes #2804

### CODE REVIEW

**Comment by veeceey** [comment]
just following up — would love some feedback on this when you get a chance

File: zellij-server/src/panes/tiled_panes/tiled_pane_grid.rs [hunk 1]
```diff
@@ -2325,31 +2326,60 @@ impl<'a> TiledPaneGrid<'a> {
 }
 
 pub fn split(direction: SplitDirection, rect: &PaneGeom) -> Option<(PaneGeom, PaneGeom)> {
+    split_with_size(direction, rect, None)
+}
```
Comment (jaeheonji) [inline zellij-server/src/panes/tiled_panes/tiled_pane_grid.rs]: It appears that it can be combined with the `split` function without creating a separate `split_with_size` function.

File: zellij-server/src/tab/mod.rs [hunk 2]
```diff
@@ -2291,6 +2296,24 @@ impl Tab {
         client_id: ClientId,
         completion_tx: Option<NotificationEnd>,
         borderless: Option<bool>,
+    ) -> Result<()> {
+        self.horizontal_split_with_size(
```
Comment (jaeheonji) [inline zellij-server/src/tab/mod.rs]: Same as `split`

File: zellij-server/src/tab/mod.rs [hunk 3]
```diff
@@ -2358,6 +2385,24 @@ impl Tab {
         client_id: ClientId,
         completion_tx: Option<NotificationEnd>,
         borderless: Option<bool>,
+    ) -> Result<()> {
+        self.vertical_split_with_size(
```
Comment (jaeheonji) [inline zellij-server/src/tab/mod.rs]: Same as `split`

**Review by jaeheonji — CHANGES_REQUESTED** [review]
First of all, thank you for submitting the PR.

I have a few suggestions regarding your current implementation. While the overall execution is great, please keep in mind that **Protocol Buffers** must be updated whenever CLI actions or options are added or removed.

Also, before pushing to the PR, please ensure you run `cargo xtask fmt` and `cargo xtask test`.

Lastly, as you may already be aware, there is a duplicate PR for this issue. I respect both contributions and will review them both to merge the better implementation. However, I will be giving slight priority to the PR(#4662) that was submitted first. I hope you understand.

**Comment by veeceey** [comment]
Thanks for the review @jaeheonji\! Makes sense — I'll merge split_with_size back into the existing split function instead of having a separate one. Will push the update.

File: zellij-server/src/panes/tiled_panes/tiled_pane_grid.rs (hunk 1 above)
Comment (veeceey) [reply zellij-server/src/panes/tiled_panes/tiled_pane_grid.rs]: Done -- merged split_with_size back into split with an optional size parameter. Same for horizontal_split and vertical_split in tab/mod.rs. Also ran cargo fmt.

**Comment by veeceey** [comment]
pushed a fix for the CI failures — the size parameter was added to horizontal_split, CliAction::NewPane, and NewPanePlacement::Tiled but I missed updating the test files. should be green now.
